content/backend/graphql-js/4-adding-a-database.md

```diff
@@ -7,189 +7,235 @@ answers: ["To increase performance of the server", "When using two APIs, the Gra
+Let's break down what's going on here:
+1. Import the `PrismaClient` constructor from the `@prisma/client` node module
+1. Instantiate `PrismaClient`
+1. Define an `async` function called `main` to send queries to the database. You will write all your queries insite this function.
+1. Call the `main` function
+1. Close the database connections when the script terminates
+
+Take a moment to re-type the query line and notice the helpful autocompletion you get after typing `prisma.` and `prisma.link.` which lets us see all of the possible models we can access and operations we can use to query that data:
 
-Once the command has finished running, the CLI writes the endpoint for the Prisma API to your `prisma.yml`. It will look similar to this: `https://eu1.prisma.sh/john-doe/hackernews-node/dev`.
+![](https://i.imgur.com/Zrrqwmo.png)
 
-The last step is to generate the Prisma client for your datamodel. The Prisma client is an auto-generated client library that lets you read and write data in your database through the Prisma API. You can generate it using the `prisma generate` command. This command reads the information from `prisma.yml` and generates the Prisma client accordingly.
+So now let's see things in action.
 
 <Instruction>
 
-Run the following command in your terminal:
+Run your new code with the following command:
 
-```bash(path=".../hackernews-node/prisma")
-prisma generate
+```bash(path=".../hackernews-node/")
+node src/script.js
 ```
 
 </Instruction>
 
-The Prisma client is now generated and located in `hackernews-node/src/generated/prisma-client`. To use the client, you can import the `prisma` instance that's exported from the generated folder. Here's some sample code that you could use in a simple Node script:
+You successfully queried the database with Prisma Client! Of course, we got an empty array back since the database is empty, so now let's add a mutation to create a new link.
 
-```js(nocopy)
-const { prisma } = require('./generated/prisma-client')
+<Instruction>
 
-async function main() {
+Type out the following lines of code yourself inside of the `main` function right above the `allLinks` query and pay close attention to the incredibly helpful autocompletion. First, it helps us understand that `.create()` is the operation we need (just scroll through the options) and then it actually shows us exactly how to construct the mutation!
 
-  // Create a new link
-  const newLink = await prisma.createLink({ 
-    url: 'www.prisma.io',
-    description: 'Prisma replaces traditional ORMs',
+```js(path=".../hackernews-node/src/script.js")
+const newLink = await prisma.link.create({
+    data: {
+      description: 'Fullstack tutorial for GraphQL',
+      url: 'www.howtographql.com',
+    },
   })
-  console.log(`Created new link: ${newLink.url} (ID: ${newLink.id})`)
+```
 
-  // Read all links from the database and print them to the console
-  const allLinks = await prisma.links()
-  console.log(allLinks)
-}
+</Instruction>
```

Comment:
@nikolasburk the reason that I decided to split up this from the original query is because I think the autocompletion is really so good and helpful and I wanted to sort of "train" them to use this autocompletion to help them understand how to write new queries/mutations themselves. But let me know if you think it's a bit dis-jointed or redundant to split it up into separate steps like this 🙂